std/array.asm

```diff
@@ -0,0 +1,16 @@
+/// This is a built-in function taking an array argument and returning
+/// the length of the array.
+/// This symbol is not an empty array, the actual semantics are overridden.
+let len = [];
```

Comment:
This is also confusing. Why is it done like this?

Reply:
If I just use `let len;`, then it will be a witness column. I have to assign something.
Although thinking about it, maybe it could work?
But in any case, since we don't have any annotations or stuff like this, it has to be weird in some way.